### ISSUE TITLE
prefactor1 outperforms prefactor3

### ISSUE THREAD

**Issue by rvweeren** [lofar-astron/prefactor#246]
I have done a careful comparison between prefactor 1 (the one used on the grid for LoTSS) and prefactor3 for LoTSS. The resulting image quality for very bright (~5+ Jy) sources is significantly worse in prefactor3 compared to prefactor1. This problem does not fully reveal itself for a field without bright sources and hence it was probably overlooked.

The resulting images show the DR2 final high-res image from the prefactor3 (left) vs prefactor1 (right). Around the very bright sources there is clear ringing visible that covers the entire image from prefactor3. I made sure I used identical settings and facet layout throughout the DR2 pipeline and also used the same TGSS model. In addition, I did a test switching the calibrator (since we observe two, one at the start, one at the end). This again shows the same problem between prefactor3 and prefactor1.

<img width="645" alt="image" src="https://user-images.githubusercontent.com/3879473/62936602-8ee61e80-bdca-11e9-95b4-82429992cdc8.png">
<img width="644" alt="image" src="https://user-images.githubusercontent.com/3879473/62939858-7ed23d00-bdd2-11e9-8d1f-3ed9c7ced982.png">

I am now looking at possible causes but before any prefactor 3 becomes the default this issue needs to be addressed. I think the problem is not related to a simple bug in prefactor but something related to the phases corrections we apply. More to follow.

**Comment by rvweeren** [comment]
Looking at it in a bit more detail. I identify three possible causes for the poor prefactor 3 performance. (1) Bandpass,  (2) XX-YY phase offset (polalign), and/or  (3) the clock. A close inspection shows that the rings error patterns are asymmetric with respect to the bright sources. That suggests a phase error so likely narrowing the problem down to the clock or polalign that are applied in prefactor3.

**Comment by adrabent** [comment]
Do you see any differences in the clock values applied in the diagnostic plots of prefactor1/3?

**Comment by rvweeren** [comment]
Yes, and they are rather large(!) for some stations. Up to 15 ns, which gives a enormous phase error across the band. In addition, the phase offsets are also quite different. Will show plots soon.

**Comment by adrabent** [comment]
Interesting.. it might be worth running losoto CLOCKTEC before correcting with polalign/FR/bandpass to see whether one can reconstruct the prefactor1 solutions. If not, there might be probably a difference in the algorithm rather than wrong corrections which alter the clock results.

**Comment by rvweeren** [comment]
Here are the plots. If you look closely at the clocks there are some > 10 ns offsets for a few remotes, which would really cause a large phase error. So at the moment I am leaning towards concluding something is wrong with the clocks...although the phase offsets are also  quite different.
![clock](https://user-images.githubusercontent.com/3879473/62941441-4e8c9d80-bdd6-11e9-88db-3515611f42aa.png)
![offset](https://user-images.githubusercontent.com/3879473/62941442-4e8c9d80-bdd6-11e9-8985-4bc9c0694652.png)

**Comment by rvweeren** [comment]
@adrabent I put the 3C196 calibrator data here (95 GB) [ftp://ftp.strw.leidenuniv.nl/pub/vanweeren/3C196cal.tar](url). Can you do the "losoto CLOCKTEC before correcting with polalign/FR/bandpass" test?

**Comment by adrabent** [comment]
Have you also used the international stations in your solve?

**Comment by rvweeren** [comment]
Yes, I used international stations.

No, I was plotting the time average (median). Does not change the results much though.

**Comment by adrabent** [comment]
I just ran prefactor3 doing the ClockTEC fitting and the results are even more different than in your case:
![clock2](https://user-images.githubusercontent.com/19146990/63019871-dccb5700-be9c-11e9-94c9-773cca1bb628.png)

Station details here:
![clock](https://user-images.githubusercontent.com/19146990/63019764-97a72500-be9c-11e9-8727-a10fbe7034fa.png)

The phase solutions look somewhat fine:
![ion_ph_polXX](https://user-images.githubusercontent.com/19146990/63019787-a5f54100-be9c-11e9-8c46-073cfdb1fe94.png)

But the phase residuals do not look overwhelmingly. Obviously there are clock residuals left in case of the International Stations:
![ion_ph-res_polXX](https://user-images.githubusercontent.com/19146990/63019811-b5748a00-be9c-11e9-9234-c2e5cfb4fcd6.png)

Why PL611HBA residuals are flagged I need to investigate.

**Comment by lmorabit** [comment]
@adrabent I believe PL611 is the station that has only half of the dipoles of the other stations, so we expect it to have lower SNR -- this might be why it's been flagged.

**Comment by rvweeren** [comment]
BTW Here is the prefactor3 plot of the clocks.
![clock](https://user-images.githubusercontent.com/3879473/63045099-ecb05e80-becf-11e9-9219-2b3aabe1a6f2.png).

@adrabent, if you pass me the H5 file I can make an easier comparison plot.

**Comment by adrabent** [comment]
Here is the h5parm file when solving only for TEC (and skipping the rest):
[cal_solutions.h5.tar.gz](https://github.com/lofar-astron/prefactor/files/3504811/cal_solutions.h5.tar.gz)

I also run the full prefactor3 on your data set now. I directly compared the clock differences (ns) of every station before and after the course of the calibration strategy. The differences are for most of the international stations severe.

DE601 |     1,1
DE602 |   11,1
DE603 |   14,5
DE604 |   24,5
DE605 |   13,4
DE609 |   15,0
FR606 | 229,7
PL610 |   42,5
PL611 | 183,4
PL612 |   62,7
SE607 |   26,5
UK608 |   32,8

For the remote stations it is more or less "fine" escept of RS210 and RS409.

RS106 |     0,1
RS205 |     0,1
RS208 |     1,9
RS210 |   16,4
RS305 |     0,4
RS306 |     0,5
RS307 |     2,0
RS406 |     0,3
RS407 |     0,7
RS409 |   21,3
RS503 |     0,3
RS508 |     1,1
RS509 |     1,3

It looks like many stations suffer from bad amplitude solutions in the last iteration of calibration:
![ion_ampAFlag_polYY](https://user-images.githubusercontent.com/19146990/63083635-82d49b00-bf49-11e9-9c72-44a0d60d083c.png)

Those are propagated to the phase solutions by default, causing large fractions of the bandwidth the be flagged:

![ion_ph_polXX](https://user-images.githubusercontent.com/19146990/63083699-b283a300-bf49-11e9-9974-a950e398dff8.png)

, which is a often big difference to what was the source from where the clock values were derived if skipping all the other steps before:
![ion_ph_polXX](https://user-images.githubusercontent.com/19146990/63083745-cd561780-bf49-11e9-8939-db60c219b7ee.png)

It is clearly seen in the phase residuals after subtracting clockTEC that something clearly went wrong:
![ion_ph-res_polXX](https://user-images.githubusercontent.com/19146990/63084049-916f8200-bf4a-11e9-91a7-38a53e609ffe.png)

In a more ideal case it should look like this:
![ion_ph-res_polXX](https://user-images.githubusercontent.com/19146990/63084120-ba901280-bf4a-11e9-8161-22855378503f.png)


Things to check are:
Are the clock values derived from a calibration run without correcting for FA/PA/bandpass more similar to those derived in prefactor1? Where do the diverging solutions seen in the clockTEC part of a full prefactor3-run come from? Can we achieve similar results if not propagating the amplitude solution flags to the phases like in prefactor1?

**Comment by rvweeren** [comment]
Here is the new comparison between @adrabent prefactor 3 test (clock-TEC only) and prefactor 1. Clocks seem a bit closer now.  Still there is a stations that differ more 10ns. (we should 'ignore' the international stations, the calibrator model is very poor so more wild fluctuations are expected which makes debugging more difficult). 

Another point I forgot to make.

- I think is clear the issue come (at least) mainly from the RS stations, giving the spacing of the error patterns.

- If we apply 10ns+ wrong clock this will cause a phase enormous error across band, even inside a single 2 MHz block. A 10ns clock error is so large it cannot be corrected for by the DR2 pipeline (as you need a very fine spacing along the freq. axis for the solutions to take it out)


![clockad](https://user-images.githubusercontent.com/3879473/63122082-38353c00-bfa6-11e9-9c0b-eefdf2be5cae.png)

**Comment by rvweeren** [comment]
I had a look at the polalign part, with the hope of ruling that out as the issue. I took instrument.h5imp_cal_PA and "phaseOrig". @adrabent , these are the input phases for the polalign step, right?

I then determined the phase offset between XX and YY and compared that to the XX-YY delay fit in losoto (the polalign step).

This is the comparison. A delay cannot explain the entire XX-YY phase offset so fitting a delay here is an approximation it seems to me (wondering why we abandoned the more agnostic approach of simply taking a median....was there a deeper reason for that?). On the other hand the difference between my median and the delay fit is quite small. 

If we compare with prefactor1 (few posts above)  the difference is larger. My speculation is that some of the phase offset can be described by the rotation term we solve in prefactor3 with DPPP and hence it does not end up in the offset (as would happen in prefactor1).

Taking all this information into account, and noting that any error we make here varies slowly with frequency the DR2 DI step should be able to correct (possible) polalign errors quite easily. Given that DR2 DI cannot correct for this the effect must change very rapidly along the freq. axis. So I think this points really to the clock values in prefactor3 (it still possible the the polalign plays a role in getting a wrong clock out).

![offsetr](https://user-images.githubusercontent.com/3879473/63125199-5ce0e200-bfad-11e9-8add-e2d050405423.png)

**Comment by rvweeren** [comment]
My ion_ph_res_polXX looks like this for 3C147 (the second calibrator for this observation). Very little flagging, so I doubt flagging is causing the issue. Also good quality solutions, in terms of S/N for all stations, including the internationals. If these plots need to show mostly "green" (?? @adrabent can you confirm?) there is something going very wrong here.
![ion_ph-res_polXX](https://user-images.githubusercontent.com/3879473/63133482-b1905700-bfc5-11e9-9478-47d51b17fea8.png)


(Note that for 3C147 I get the same error pattern in the DR2 LoTSS image so also the clocks for 3C147 seems to be incorrect).

**Comment by adrabent** [comment]
>  @adrabent , these are the input phases for the polalign step, right?
That's right. It is basically just a backup of the input phases.

> If these plots need to show mostly "green" (?? @adrabent can you confirm?)
'Green' simply means there are basically no residuals. If you have red/blue or whatever different color you still have (after all corrections) an absolute phase offset from zero left. For our clock issue it is even more relevant that the residuals do not show a frequency-dependent behaviour. In your last plot there is still a clear gradient in frequency visible for RS210HBA and the international stations, which means there is still a significant phase drift left (and thus not properly corrected for).

**Comment by rvweeren** [comment]
I've been investigating in more detail the clock-TEC  fitting. I found that the input phases for the clock-TEC fitting are not good. Top: the phases at the start in prefactor. Right: the phases clock-TEC fitting will works. I can easily see how you get wrong clocks out if you try to fit this. I suspect that the part where it deals with the bandpass/flagging(?) is to blame, but it is not for me to debug this as I am not familiar with these steps.

<img width="182" alt="image" src="https://user-images.githubusercontent.com/3879473/63216522-c809f000-c136-11e9-8c95-e7ee73f295f0.png">

<img width="183" alt="image" src="https://user-images.githubusercontent.com/3879473/63216567-22a34c00-c137-11e9-9c34-a5c434bad2f2.png">

**Comment by rvweeren** [comment]
@adrabent, @revoltek  The polign and FR look fine. How can we end up with these weird "corrupted" phases? The FR removal propagates weirdness (from the amplitudes) to the phases?

**Comment by adrabent** [comment]
In order to have a more fair comparison with prefactor1 I re-run the calculation without International Stations (since pref1 removes them completely). Since the bandpass results are so extremely bad for the international stations, applying these solutions and then trying the re-calibrate with a significant amount of heavily corrupted baselines will for sure alter the quality of the calibration results.

These are the clock results of a fully prefactor3 run, excluding International Stations:

Phase solutions (for XX):
![ion_ph_polXX](https://user-images.githubusercontent.com/19146990/63267273-f223f500-c291-11e9-8338-e026f8bbf1ad.png)

Clock fits:
![clock](https://user-images.githubusercontent.com/19146990/63267194-cc96eb80-c291-11e9-8ce2-5181230f63d1.png)

And the residuals look pretty fine (even if not perfect):
![ion_ph-res_polXX](https://user-images.githubusercontent.com/19146990/63267241-e20c1580-c291-11e9-9b1d-d18b910bdd90.png)

I compared the clock offsets with the calibration and the difference is basically present for all Remote Stations:

RS106 |  2,9
RS205 |  3,7
RS208 |  1,3
RS210 |  11,6
RS305 |  4,4
RS306 |   4,9
RS307 |    2,6
RS406 |  4,2
RS407 |   3,1
RS409 |  14,8
RS503 |   3,4
RS508 |  1,4
RS509 |   3,7

@rvweeren 
Please feel free to compare these results with the pref1 outcome:
[clock.h5.tar.gz](https://github.com/lofar-astron/prefactor/files/3515997/clock.h5.tar.gz)

**Comment by rvweeren** [comment]
@adrabent, Yes you make a good point about the international baselines and bad amplitudes. I will do the comparison with your new clocks and the prefactor 1 clocks.

**Comment by rvweeren** [comment]
Here is the full comparison plot. Now results are in much better agreement for 3C196.
I also included 3C147. 3C147 is almost point-like also for the international stations. For 3C147, for the Dutch stations (international were included in the solve), the results between prefactor 3 and my manual fit (using the prefactor 1 script) agree quite well (apart for one station).

(3C147 was observed at quite a different time than 3C196 so that's why in general the clocks could be rather different)

![clockcom](https://user-images.githubusercontent.com/3879473/63279784-ec3a0e00-c2a9-11e9-8a67-a90e51248b6a.png)


((I do observe one clear issue, that is losoto clock-TEC fitting is not performing well for international stations. This is a separate issue/bug. I'll make a new ticket for that one.))

**Comment by adrabent** [comment]
I modified the current prefactor strategy and applied the reference bandpasses for the International Stations already after the regular bandpass step so that these corrections were already available in the ionosphere step.

The input phase solutions look basically identical to the pref3 run, not using IS, and the pref3 run without correcting for PA, FR, and bandpass (see above posts)
![ion_ph_polXX](https://user-images.githubusercontent.com/19146990/63328039-b17ab900-c32f-11e9-8540-288f4f649d05.png)

The new clock results:
![clock](https://user-images.githubusercontent.com/19146990/63328094-ca836a00-c32f-11e9-9ea2-7ce7ca2d3f25.png)

The residual is cleaner (but not perfect), except for IS:
![ion_ph-res_polXX](https://user-images.githubusercontent.com/19146990/63328106-d4a56880-c32f-11e9-9fd5-748d6ec45af5.png)

Still the clock fits are slightly different!
Compared to the pref3 run, skipping PA, FR, and bandpass:
DE601 |   13,4
DE602 |   43,0
DE603 |   26,7
DE604 |   16,9
DE605 |   11,2
DE609 |   1,9
FR606 |  26,3
IE613 |   54,2
PL610 |  34,2
PL611 |  199,2
PL612 |   34,6
RS106 |  3,0
RS205 |  3,4
RS208 |  1,2
RS210 |  0,8
RS305 |  3,8
RS306 |   4,0
RS307 |   3,5
RS406 |  4,3
RS407 |   5,2
RS409 |   6,1
RS503 |    3,8
RS508 |    6,1
RS509 |   7,3
SE607 |    20,1
UK608 |  22,2

Compared to the full pref3 run, removing International Stations:
RS106 |  0,2
RS205 |  0,2
RS208 |  2,0
RS210 |  5,6
RS305 |  0,3
RS306 |  0,4
RS307 |  1,1
RS406 |  0,4
RS407 |  1,4
RS409 |  0,4
RS503 |  0,1
RS508 |  3,7
RS509 |  4,8

At least from a visual impression, the input phase solutions look more or less identical. I am still surprised that the clock fits deviate so much from each other. Here you find the solutions H5parm:

[clock_modified_pref3.h5.tar.gz](https://github.com/lofar-astron/prefactor/files/3519307/clock_modified_pref3.h5.tar.gz)

**Comment by lmorabit** [comment]
@rvweeren @adrabent I would say that some of this might be model dependent since 3C 147 has clock values that are more similar with and without including the international stations in the solve. At HBA frequencies, 3C 147 does have a bit of structure, but it's contained within about 1.5" and the second component is much fainter. 

Where is the data?  I have a bit of time and could test solving directly for clock/TEC with ddecal and/or gaincal rather than separating them in solution space.

**Comment by adrabent** [comment]
https://github.com/lofar-astron/prefactor/issues/246#issuecomment-520828337

**Comment by rvweeren** [comment]
@lmorabit, yes you are right, this is should be model dependent. So for 3C147 or 3C48 the trick of @adrabent using a reference bandpass for the international stations would not be needed.

**Comment by lmorabit** [comment]
@rvweeren where is the data for 3C 147 you've been working on?

**Comment by rvweeren** [comment]
@lmorabit here it is: ftp://ftp.strw.leidenuniv.nl/pub/vanweeren/3C147.tar

**Comment by adrabent** [comment]
@rvweeren 
I slightly changed the strategy of the transfer of reference solutions. Would it be possible for you to re-run the problematic calibrator with these changes in order to check whether the artifacts in the final DR2 image are still there?

**Comment by rvweeren** [comment]
Thanks @adrabent, will do that (@maaijke is also working on updating the clock-TEC fitting since I suspect that was also partly to blame)

**Comment by adrabent** [comment]
@maaijke 

I tested your new implementation of the clockTEC seperation, but the results did not improve for the International Stations. The only thing which appeared to change is the overall phase offset, see https://github.com/lofar-astron/prefactor/issues/246#issuecomment-522901562

![ion_ph-res_polXX](https://user-images.githubusercontent.com/19146990/64154538-61245080-ce31-11e9-9758-3a8c568d9cae.png)

**Comment by maaijke** [comment]
ok, can I test these data on cep?

**Comment by adrabent** [comment]
@maaijke 
https://github.com/lofar-astron/prefactor/issues/246#issuecomment-522911998

**Comment by maaijke** [comment]
I prefer to start from the phase solutions....\

**Comment by adrabent** [comment]
ftp://kso.tls-tautenburg.de/pub/alex/3C196_ion.h5.tar.gz

**Comment by rvweeren** [comment]
I verified that the clock values that I now apply in prefactor3 (for now using my own clock-TEC fitting code overwriting the losoto computed values) are within a few nano second of the prefactor 1 solutions. Now running the DR2 pipeline (since a few days) to see if the artifacts are gone. Will take a 1-2 more days to see the results. That should tell -if- the wrong clock values were really to blame. (clearly the clock values were not correct in prefactor 3, -but- if that caused the reduced image quality was not fully proven, there still is the possibility of the phase offsets are also causing problems)

**Comment by rvweeren** [comment]
@adrabent, @maaijke We should keep in mind that 

1. for the international stations the 3C196 4-component model used is way too simplistic(!). Therefore I do not think we can require that the correct clocks need to come out correctly in this case for the international stations. What we want to see is that at least for the Dutch RS (but including the international stations in the solves in prefactor3) the clock values found by losoto are now correct (and not affected by the simplistic 3C196 model).

2. The clock values between prefactor 1 (my code) and losoto, with the new updates @maaijke  included, agree for the international stations -if- we use a very high quality model (that is appropriate for the long baselines). 


(@adrabent If I am correct the results/plot you posted a few hrs ago used a low-res 4-component 3C196 model and you cannot draw conclusion about the clock-TEC fitting for the international stations from that, if you want to do that you need to use a much higher/better resolution source model)

**Comment by rvweeren** [comment]
Less good news, the clock is not the only problem it seems in prefactor3, there are more (still unknown) issues beyond that. Left prefactor3 updated clock, middle old prefactor3, right prefactor1. So the new clock does improve the image quality a bit, but as much as a had hoped. I am now turning my attention to the phase-offsets again (maybe fitting a simple delay is not appropriate). I will overwrite the prefactor3 offsets (xx-yy delays) with the prefactor1 offsets and see what that does....will take another week+ because I need to redo the calibrator part (with the new offsets) and the entire DR2 pipeline....

<img width="683" alt="image" src="https://user-images.githubusercontent.com/3879473/64326025-7fbd4f80-cfc9-11e9-9d19-02364d9abed9.png">

**Comment by twshimwell** [comment]
Ah what a pain... have you tried directly comparing the ddf-pipeline solutions for this facet to see if that gives any insight?

**Comment by twshimwell** [comment]
ps the data that comes from grid processing goes through a very old prefactor:

https://github.com/lofar-astron/prefactorusing commitdd68c57

**Comment by rvweeren** [comment]
@twshimwell this is not the only facet that shows problem. It's everywhere (not surprising since it a DI problem introduced by prefactor3).

**Comment by rvweeren** [comment]
Here is a careful comparison between the xx-yy offsets with some help from @wllwen007. Right, my own time median (from prefactor3 phases), middle prefactor3, right prefactor1. My phases agree very well with those from prefactor 1 (note there is smoothing in prefactor 1 which I did not do in my case hence the noiser solutions). The zoom version shows fitting xx-yy fitting with a delay-only terms  (as happens in prefactor 3) is an approximation and it does not capture the full picture. On the other hand the differences are not really large (in radians). It would surprise me that this could degrade image quality. But I want to test it to be 100% sure. (if this does not solve the issue only the bandpass is left as a possibility)

![c1](https://user-images.githubusercontent.com/3879473/64453380-b2aa3500-d0e8-11e9-979b-2ed3ce4cb890.png)
![c2](https://user-images.githubusercontent.com/3879473/64453388-b8077f80-d0e8-11e9-85a7-cb94b6deb75e.png)

**Comment by rvweeren** [comment]
With smoothing implemented

![c1](https://user-images.githubusercontent.com/3879473/64454120-6d870280-d0ea-11e9-9717-725b91aaa7ee.png)

**Comment by rvweeren** [comment]
Need about two more days to be 100% sure but it looks like the phase offset does not significantly alter the image quality. Now started looking at the bandpasses (that is the only major option left on my list of possibilities, unless I overlook something...).
![bandpass](https://user-images.githubusercontent.com/3879473/64724121-25912280-d4d2-11e9-8369-96096f5908c6.jpg)

The main differences between prefactor 1 and 3 are small wiggles. In prefactor 1 we "smooth" these out.

**Comment by rvweeren** [comment]
Tiny improvement with the prefactor1 XX-YY phase offsets (with respect to the ones from prefactor3). That leaves the bandpass. Will now replace the bandpasses from prefactor3 with prefactor1....

**Comment by rvweeren** [comment]
Tests are completed, problem remains. This suggest the problem is somewhere else, maybe in the target pipeline (something with the order of the applycal and station beam?).  Not sure how the prefactor 1 target part precisely differs from prefactor3....

**Comment by rvweeren** [comment]
I might be onto something, this is a view of some of the output of the target pipeline, that does into the DR2 pipeline.

Most important(!) the SB at the darker were never observed by LOFAR. So prefactor3 is making up fake data here ?? That should --never-- happen. Only the "light" band are observed SB. The rest seems to be made up by prefactor3.

<img width="699" alt="image" src="https://user-images.githubusercontent.com/3879473/65589074-c2e75e80-df88-11e9-983a-3955c77b32a2.png">

**Comment by rvweeren** [comment]
Another more extreme example. Only one SB at 175.8 MHz was observed, but prefactor manages to produce fake data above and below it....

<img width="692" alt="image" src="https://user-images.githubusercontent.com/3879473/65590118-8452a380-df8a-11e9-93eb-b1db69e8a0dd.png">

**Comment by rvweeren** [comment]
And another example at 164.3 MHz. This SB was never observed but unflagged fake data is created. (this is what I feed into the DR2 pipeline....this is certainly a recipe for troubles)

<img width="693" alt="image" src="https://user-images.githubusercontent.com/3879473/65590583-51f57600-df8b-11e9-8d52-0dd2df6f9582.png">

**Comment by rvweeren** [comment]
Probably related to the interpolation steps.

https://github.com/lofar-astron/prefactor/blob/2223416d05fc74d3bff6b93ec58fea219d9f24b3/Pre-Facet-Target.parset#L315

https://github.com/lofar-astron/prefactor/blob/2223416d05fc74d3bff6b93ec58fea219d9f24b3/Pre-Facet-Target.parset#L479

**Comment by tikk3r** [comment]
Another oddity, if those are indeed missing subbands, Is that that data is not flagged (as the dummy data inserted by DPPP should be 0 and flagged).

**Comment by rvweeren** [comment]
Here is a prefactor1 (left) vs prefactor3 (right) comparison where one SB was missing. 


(@tikk3r I think because it interpolates from data where there is no bandpass applied to yet, but not fully sure. I also remember your 168 MHz block looked a bit "weird" for the LB LH dataset having a banded structure, that's probably the same issue we're seeing, there was some interpolated data there)

<img width="562" alt="image" src="https://user-images.githubusercontent.com/3879473/65630303-770ed680-dfd5-11e9-8a9e-e27b1363496f.png">

**Comment by darafferty** [comment]
OK, I'm pretty sure you're right -- it is the interpolation that's creating the fake data. I will remove these steps. (Perhaps later we can revisit whether they are beneficial in some cases. If I recall correctly, we did see some improvement when interpolation is combined with averaging, but obviously we don't want to create fake data!)

**Comment by rvweeren** [comment]
@darafferty, thanks will test. I took the interpolation part out manually myself yesterday evening  (a quick hack) and that target pipeline is running. Should be finished tomorrow.

**Comment by rvweeren** [comment]
Confirmed that taking the interpolating out fixes the "fake" data problem. Now running the DR2 pipeline to verify this fixes the dynamic range issue (will take a week).

**Comment by rvweeren** [comment]
Good news. Taking out the interpolation fixes the main problem (left without interpolation, right with).

(DDF pipeline is still running. At this stage, the prefactor1 final image still looks better, but looking at an intermediate stage image. A few more days and will know if also the last artifacts around the bright sources are cleaned up....)

<img width="565" alt="image" src="https://user-images.githubusercontent.com/3879473/65867264-62846280-e376-11e9-8e2a-b859ee1fd187.png">

**Comment by rvweeren** [comment]
Final confirmation, problem is fixed now. Happy to finally close this thread.

To summarize what was uncovered:
- the interpolation step creates unflagged 'fake' data which causes major degradation in image quality 
- the clock-TEC fitting in losoto was flawed, being inferior to what we did in prefactor1, causing up to 30ns clock errors (this should be fixed in the latest losoto but we should keep an eye on it as it has not been tested that much yet).
- if international baselines are included in the solve with a poor quality model, it results in flagging of data and thus poor end products (because the bandpass solutions are severely compromised). A solution for that was added by @adrabent that will use a good 'external' bandpass solution set.
